File: cpp/H1 - THREAD - STL - TOMBOL/spaceList.cpp
#include "spaceList.h"

SpaceList::SpaceList()
{
}

SpaceList::~SpaceList()
{
}

void SpaceList::getObject (int n, Shape& spaceObject)
{
    spaceObject = listOfObject[n];
}

void SpaceList::setObject(Shape S)
{
    listOfObject.push_back(S);
}

int SpaceList::getLength()
{
    return listOfObject.size();
}

Shape SpaceList::getShape (int id)
{
    bool found = false;
    int i =0;

    while ((!found) && (i< listOfObject.size()))
    {
        if (listOfObject[i].getID() == id)
        {
            found = true;
        }
        else
        {
            i++;
        }

    }
    return listOfObject[i];
}

SpaceList& SpaceList::deleteList (Shape checkObject)
{
    bool found = false;
    int i = 0;
    while ((!found) && (i < listOfObject.size()))
    {
        if (checkObject.getID() == listOfObject[i].getID())
        {
            found = true;
            listOfObject.erase(listOfObject.begin()+i);
        }
        else
        {
            i++;
        }
    }
    return *this;
}

SpaceList& SpaceList::BigBang()
{
    listOfObject.clear();
    return *this;
}


void SpaceList :: setList(int i, Shape X)
{
    listOfObject[i] = X;
}

File: java/13514022_ITI/src/spacelist/SpaceList.java
package spacelist;
import java.util.ArrayList;
import makhluk.Shape;

/** Kelas SpaceList. Kelas objek yang menampung makhluk yang sedang hidup./
 * @author Kristianto K./13514075
 */
public class SpaceList  {
    /** vector yang merepresentasikan list dari objek yang sedang hidup. */
    private ArrayList<Shape> listOfObject;

    /** Ctor spacelist.
    */
    public SpaceList() {
        this.listOfObject = new ArrayList<>();
    }
    /** Function untuk menghapus objek x dari spacelist.
    *
    * @param x objek yang dihapus dari list
    * @return Spacelist
    *
    */
    public final SpaceList deleteList(final Shape x) {
        boolean found = false;
        int i = 0;
        while ((!found) && (i < listOfObject.size())) {
            if (x.getID() == listOfObject.get(i).getID()) {

                found = true;
                listOfObject.remove(i);
            } else {
                i++;
            }
        }
        return this;
    }
    /** Function yang mengembalikan objek ber-ID id dari list.
    *
    * @param id ID dari objek yang ingin diambil
    * @return Shape
    *
    */
    public final Shape getShape(final int id) {
        boolean found = false;
        int i = 0;
        while ((!found) && (i < listOfObject.size())) {
            if (listOfObject.get(i).getID() == id) {
                found = true;
            } else {
                i++;
            }
        }
        return listOfObject.get(i);
    }
    /** Method untuk menambahkan objek baru di spacelist.
    * Penambahan dilakukan setelah objek terakhir pada list.
    *
    * @param s shape yang di tambahkan pada list
    *
    */
    public final void setObject(final Shape s) {
        listOfObject.add(s);
    }
    /** Function untuk mengembalikan panjang list.
    *
    * @return int panjangList
    *
    */
    public final int getLength() {
        return listOfObject.size();
    }
    /** Method untuk mendapatkan objek dari list sesuai indeks.
    *
    * @param i indeks objek pada list
    * @return Shape makhluk pada indeks ke-i
    *
    */
    public final Shape getObject(final int i) {
        Shape x;
        x = listOfObject.get(i);
        return x;
    }
    /** Setter objek dengan indeks i, menjadi bernilai X.
    *
    * @param i indeks
    * @param s shape yang melakukan assign
    *
    */
    public final void setList(final int i, final Shape s) {
        listOfObject.set(i, s);
    }
    /** Menghapus seluruh makhluk dari list.
    *
    * @return Spacelist yang sudah dihapus
    *
    */
    public final SpaceList bigBang() {
        listOfObject.clear();
        return this;
    }
    /** Mengecek apakah spacelist isEmpty.
    *
    * @return true jika kosong dan false jika ada
    *
    */
    public final boolean isEmpty() {
        return listOfObject.isEmpty();
    }
}
